src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/utils/FileUtils.java

```diff
@@ -26,4 +28,23 @@ public static boolean isWordDocument(CaseDocument caseDocument) {
         return isDocumentType(caseDocument, ".DOC", ".DOCX");
     }
 
+    /**
+     * Inserts a timestamp into a given filename.
+     *
+     * @param filename the original filename (e.g., "ABC.pdf")
+     * @return the filename with the timestamp inserted (e.g., "ABC_20241209_123456.pdf")
+     * @throws IllegalArgumentException if the filename is invalid
+     */
+    public static String insertTimestamp(String filename) {
+        if (filename == null || !filename.contains(".")) {
+            throw new IllegalArgumentException("Invalid filename: " + filename);
+        }
+
+        String timestamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        int lastDotIndex = filename.lastIndexOf(".");
+        String baseName = filename.substring(0, lastDotIndex);
+        String extension = filename.substring(lastDotIndex);
+
+        return baseName + "_" + timestamp + extension;
+    }
```

Comment:
Just in case somebody gives you filename.txt.txt, or somethin similar, consider changing the check so that its confirms you only have 1 "."  in the filename.

Reply:
[ee9173b](https://github.com/hmcts/finrem-case-orchestration-service/pull/2021/commits/ee9173be2eb69871a854c0b9611383b192d6d327)
I think it's already handled.